frontend/src/pages/document.tsx

```diff
@@ -61,7 +113,18 @@ export function DocumentPage({
             label="back to document gallery"
             onClick={() => navigate('/')}
           />
-          <TopBarTitle>{data?.name}</TopBarTitle>
+          <DocumentTitle
+            name={data?.name}
+            onChange={(newTitle) => {
+              mutate({ ...data, name: newTitle });
+              updateDocument({ document_id: documentId, name: newTitle })
+                .catch((e) => {
+                  console.error(e);
+                  mutate(data);
+                }) // reset to old name
+                .then(() => mutate());
+            }}
+          />
```

Comment:
This should only be rendered if the `data` was loaded already. Otherwise we end up with just a pencil icon without a document name

```suggestion
          {data !== undefined ? <DocumentTitle
            name={data?.name}
            onChange={(newTitle) => {
              mutate({ ...data, name: newTitle });
              updateDocument({ document_id: documentId, name: newTitle })
                .catch((e) => {
                  console.error(e);
                  mutate(data);
                }) // reset to old name
                .then(() => mutate());
            }}
          /> : <></>}
```